src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -698,4 +642,158 @@ public static void constructLobs(Random random, IonType type, IonWriter writer)
                 throw new IllegalStateException ("Please provide CLOB or BLOB");
         }
     }
+
+    /**
+     * Write Ion structs which conform with the constraints in Ion schema.
+     * @param format is the format of the generated file, select from set (ion_text | ion_binary).
+     * @param size specifies the size in bytes of the generated file.
+     * @param path the destination of the generated file.
+     * @param fields is an Ion struct which contains the constraints needed during data generating process.
+     * @throws Exception if errors occur when writing data.
+     */
+    public static void writeRandomStructValues(int size, String format, String path, IonStruct fields, IonList annotations) throws Exception {
+        File file = new File(path);
+        try (IonWriter writer = WriteRandomIonValues.formatWriter(format, file)) {
+            WriteRandomIonValues.writeRequestedSizeFile(size, writer, file, IonType.STRUCT,fields, annotations);
+            WriteRandomIonValues.printInfo(path);
+        }
+    }
+
+    /**
+     * Construct Ion structs based on the provided constraints.
+     * @param fields is an Ion struct which contains the constraints needed during the data generating process.
+     * @param annotations is an Ion list which contains the required annotations.
+     * @param writer writes Ion struct data.
+     * @throws IOException if errors occur when writing data.
+     */
+    public static void constructIonStruct(IonStruct fields, IonList annotations, IonWriter writer) throws IOException {
+        Random random = new Random();
+        try (IonReader reader = IonReaderBuilder.standard().build(fields)) {
+            reader.next();
+            reader.stepIn();
+            int index = random.nextInt(annotations.size());
+            writer.addTypeAnnotation(String.valueOf(annotations.get(index)));
+            writer.stepIn(IonType.STRUCT);
+            while (reader.next() != null) {
+                IonSystem system = IonSystemBuilder.standard().build();
+                String fieldName = reader.getFieldName();
+                IonValue struct = system.newValue(reader);
+                IonStruct value = (IonStruct) struct;
+
+                // if the value of "occurs" is optional, the integer represents this value is 1 or 0.
+                int occurTime = WriteRandomIonValues.parseConstraints(value, "occurs");
+                if (occurTime == 0) {
+                    continue;
+                }
+                writer.setFieldName(fieldName);
+                IonType type = null;
+                type = IonType.valueOf(value.get("type").toString().toUpperCase());
+                switch (type) {
+                    // if more types of Ion data are available, the logic should be added below.
+                    case STRING:
+                        String codepointRange = "[0, 1114111]";
+                        List<Integer> range = parseRange(codepointRange);
```

Comment:
This range looks like it should be a constant.

Reply:
> This range looks like it should be a constant.

This part is updated as well.